pyvo/utils/prototype.py

```diff
@@ -0,0 +1,191 @@
+import inspect
+import warnings
+from dataclasses import dataclass
+from functools import wraps
+from typing import Dict, Iterable
```

Comment:
I'd like the idea of using type hints. If we do this we should probably also add static checkers to the ci (outside the scope of this PR)